### PR TITLE
Update for OpenSSL 1.1 API.

### PR DESCRIPTION
**Pull request by bobfeldbauer**
Similar to https://github.com/bitcoin/bitcoin/pull/9326, but modified for Dash accordingly.

### CODE REVIEW

**Comment by UdjinM6** [comment]
Thanks, looks good!

Shouldn't `src/qt/paymentrequestplus.cpp ` be patched as well though?

**Comment by bobfeldbauer** [comment]
Yes. I added a second commit to also fix src/qt/paymentrequestplus.cpp now.

**Comment by taw00** [comment]
Awesome. Thanks! I guess my prod of flare paid off. :)

File: src/qt/paymentrequestplus.cpp [hunk 1]
```diff
@@ -159,15 +159,28 @@ bool PaymentRequestPlus::getMerchant(X509_STORE* certStore, QString& merchant) c
         std::string data_to_verify;                     // Everything but the signature
         rcopy.SerializeToString(&data_to_verify);
 
-        EVP_MD_CTX ctx;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+	EVP_MD_CTX *ctx = EVP_MD_CTX_new();
```
Comment (thelazier) [inline src/qt/paymentrequestplus.cpp]: This line will look perfect if add a space at the beginning.

Comment (bobfeldbauer) [inline src/qt/paymentrequestplus.cpp]: I don't see what you mean. Can you elaborate?

Comment (thelazier) [inline src/qt/paymentrequestplus.cpp]: Indentation.

Comment (bobfeldbauer) [inline src/qt/paymentrequestplus.cpp]: I assume you mean changing the tab at the beginning of line 163 to spaces?

**Pull request merged**